ERCS/erc-7546.md

```diff
@@ -0,0 +1,187 @@
+Implement `supportsInterfaces()` to return a list of registered interfaceIDs.
+```solidity
+function supportsInterfaces() public view returns (bytes4[] memory);
+```
+
+##### 3.1.2 Implementation Contracts:
+Since the Proxy MUST forward all calls (CALL or STATICCALL) to the Dictionary, the implementation for ERC-165 is RECOMMENDED to be registered in the Dictionary's mapping instead.
+
+Furthermore, it is RECOMMENDED for each implementation contract to implement ERC-165's `supportsInterface(bytes4 interfaceID)` to ensure that it correctly implements the function selector being registered when added to the Dictionary.
+
+#### 3.2 For off-chain support:
+Even though the Proxy doesn't present an external interface, one can determine the Dictionary address that controls the actual behavior and retrieve a list of function selectors associated with registered implementations.
+
+1. Obtain the address of the Dictionary, which manages the implementation contracts defining the actual behavior, from the Proxy's specific slot (section 4) using an `eth_getStorageAt()` JSON-RPC request.
+2. Use the `supportsInterfaces()` interface of the Dictionary to retrieve a list of function selectors corresponding to the implementation contracts registered in the Dictionary.
+3. Utilize the `getImplementation(bytes4 functionSelector)` function of the Dictionary to obtain the actual implementation address.
+If the implementation contracts managed by the Dictionary corresponding to the Proxy are properly verified, tools like Etherscan can verify the behavior of the Proxy.
+
+### 4. Storage Layout
+The Proxy shares storage with several implementation contracts, making it prone to storage conflicts when using a sequential slot allocation starting from slot 0.
+
+Storage MUST be managed properly. The matter of storage management techniques has been a subject of debate for years, both at the ERC level and the language level. However, there is still no definitive standard. Therefore, this ERC does not go into the specifics of storage management techniques.
+
+It is RECOMMENDED to choose the storage management method that is considered most appropriate at the time.
+
+For instance, the Dictionary address is stored in accordance with the method defined in ***[ERC-1967: Proxy Storage Slots](./eip-1967.md)***, as follows, and other storage is arranged according to useful storage layout patterns, such as ***[ERC-7201: Namespaced Storage Layout](./eip-7201.md)***.
+
+
+## Rationale
+- ### Separating the Dictionary and Proxy contracts:
+  The separation of the Dictionary from the Proxy was driven by two primary motivations:
+
+  1. Enabling the cloning of a Proxy with an upgradeable common implementation template.
+  2. Allowing the upgrade of implementations for all Clones simultaneously.
+
+  To achieve these goals, the implementation addresses were externalized as the Dictionary instead of including them within the Proxy, a concept akin to the Beacon Proxy approach.
```

Comment:
What is the "Beacon Proxy" approach?

Reply:
The Beacon Proxy approach, as described in ERC1967, is an approach that externalizes the storage of the implementation address from the Proxy contract to a Beacon contract. By switching the implementation address stored in the Beacon, it becomes possible to change the behavior of all Proxies at once.